.github/workflows/docs.yaml

```diff
@@ -0,0 +1,29 @@
+on:
+  push:
+    branches: [main]
+
+jobs:
+  build-deploy:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@master
+    - name: Install nightly rust toolchain
+      uses: actions-rs/toolchain@v1
+      with:
+        profile: minimal
+        # Needed for use of '-Zunstable-options' since '--enable-index-page' is unstable
+        toolchain: nightly
+        override: true
+    - name: Build docs
+      uses: actions-rs/cargo@v1
+      env:
+       RUSTDOCFLAGS: "--enable-index-page -Zunstable-options"
+      with:
+        command: doc
+        args: --no-deps --workspace --exclude xtask --features compat,unstable-pre-spec,unstable-synapse-quirks
```

Comment:
One more cargo option we should use: [`-Zrustdoc-map`](https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#rustdoc-map). And `.cargo/config` has to be updated accordingly, but that's super easy.

Reply:
And for the crate features, I think this is the list we should use: `api,events,signatures,appservice-api,client-api,federation-api,identity-service-api,push-gateway-api,either,rand,markdown,compat,unstable-pre-spec`. Most of these are for the `ruma` crate; `--all-features` would be preferable but can't be used right now due to the TLS crate features of `ruma-client` being mutually exclusive.

Reply:
We could define a `docs` feature that enables the necessary features, like in matrix-rust-sdk. It's easier to use locally in the command line and can be reused to launch all tests.

Reply:
I don't think that's necessary. I'm planning to have `--all-features` work for ruma-client at some point.

Reply:
The rustdoc-map flag and new features are all added now